rpc/client.go

```diff
@@ -232,6 +232,15 @@ func (c *Client) loop() {
 			c.handleResponse(resp)
 			if c.err != nil {
 				logrus.Infof("Exiting rpc loop for %v with err %v", c.peerAddr, c.err)
+				if err := c.wire.CloseRead(); err != nil {
+					logrus.Error("failed to close read, error: ", err)
+				}
+				if err := c.wire.CloseWrite(); err != nil {
+					logrus.Error("failed to close write, error: ", err)
+				}
+				if err := c.wire.Close(); err != nil {
```

Comment:
We can't be sure that other GO routines are exited. So, doing Close here can be risky as the same fd can be re-assigned when other GO routines are still using it.